### ISSUE TITLE
[0.11-nightly] COUNT panics when passed a string field

### ISSUE THREAD

**Issue by beckettsean** [influxdata/influxdb#5823]
```
InfluxDB shell 0.11.0.n1455609633
> use mydb
> insert strings value="foo"
> insert strings value="faf"
> insert strings value="faa"
> select * from strings
name: strings
-------------
time			value
1456350036257344100	foo
1456350039327375433	faf
1456350042087391311	faa

> select count(value) from strings
ERR: Get http://localhost:8086/query?db=mydb&epoch=ns&q=select+count%28value%29+from+strings: EOF
```

```
[query] 2016/02/24 16:40:57 SELECT count(value) FROM mydb."default".strings
panic: unsupported count iterator type: *influxql.stringMergeIterator

goroutine 308981 [running]:
github.com/influxdata/influxdb/influxql.newCountIterator(0x7f39ad40cc18, 0xc20ede14a0, 0x7f39b410d530, 0xc215091a10, 0x105b660, 0x0, 0x0, 0xc2118ac870, 0x1, 0x1, ...)
        /tmp/tmp.W3SuAPdv1O/src/github.com/influxdata/influxdb/influxql/call_iterator.go:67 +0x5f7
github.com/influxdata/influxdb/influxql.NewCallIterator(0x7f39ad40cc18, 0xc20ede14a0, 0x7f39b410d530, 0xc215091a10, 0x105b660, 0x0, 0x0, 0xc2118ac870, 0x1, 0x1, ...)
        /tmp/tmp.W3SuAPdv1O/src/github.com/influxdata/influxdb/influxql/call_iterator.go:41 +0x155
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).CreateIterator(0xc21c75c000, 0x7f39b410d530, 0xc215091a10, 0x105b660, 0x0, 0x0, 0xc2118ac870, 0x1, 0x1, 0x0, ...)
        /tmp/tmp.W3SuAPdv1O/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:663 +0x262
github.com/influxdata/influxdb/tsdb.(*Shard).CreateIterator(0xc212b7de40, 0x7f39b410d530, 0xc215091a10, 0x105b660, 0x0, 0x0, 0xc2118ac870, 0x1, 0x1, 0x0, ...)
        /tmp/tmp.W3SuAPdv1O/src/github.com/influxdata/influxdb/tsdb/shard.go:387 +0x9a
github.com/influxdata/influxdb/tsdb.func·007(0x0, 0x0)
        /tmp/tmp.W3SuAPdv1O/src/github.com/influxdata/influxdb/tsdb/shard.go:440 +0xcb
github.com/influxdata/influxdb/tsdb.Shards.CreateIterator(0xc213cbef40, 0x3, 0x3, 0x7f39b410d530, 0xc215091a10, 0x105b660, 0x0, 0x0, 0xc2118ac870, 0x1, ...)
        /tmp/tmp.W3SuAPdv1O/src/github.com/influxdata/influxdb/tsdb/shard.go:447 +0x1c3
github.com/influxdata/influxdb/tsdb.(*Shards).CreateIterator(0xc213cbefa0, 0x7f39b410d530, 0xc215091a10, 0x105b660, 0x0, 0x0, 0xc2118ac870, 0x1, 0x1, 0x0, ...)
        <autogenerated>:68 +0xfd
github.com/influxdata/influxdb/influxql.buildExprIterator(0x7f39b410d530, 0xc215091a10, 0x7f39b4208c50, 0xc213cbefa0, 0x7f39b410d530, 0xc215091a10, 0x105b660, 0x0, 0x0, 0xc2118ac870, ...)
        /tmp/tmp.W3SuAPdv1O/src/github.com/influxdata/influxdb/influxql/select.go:226 +0x1a7d
github.com/influxdata/influxdb/influxql.func·034(0x0, 0x0)
        /tmp/tmp.W3SuAPdv1O/src/github.com/influxdata/influxdb/influxql/select.go:151 +0x64c
github.com/influxdata/influxdb/influxql.buildFieldIterators(0xc20802f060, 0x1, 0x1, 0x7f39b4208c50, 0xc213cbefa0, 0x0, 0x0, 0x105b660, 0x0, 0x0, ...)
        /tmp/tmp.W3SuAPdv1O/src/github.com/influxdata/influxdb/influxql/select.go:187 +0x103
github.com/influxdata/influxdb/influxql.Select(0xc2181b2dd0, 0x7f39b4208c50, 0xc213cbefa0, 0xc214a87b68, 0x0, 0x0, 0x0, 0x0, 0x0)
        /tmp/tmp.W3SuAPdv1O/src/github.com/influxdata/influxdb/influxql/select.go:77 +0xd7a
github.com/influxdata/influxdb/tsdb.(*QueryExecutor).PlanSelect(0xc208030d70, 0xc2181b2dd0, 0x2710, 0x0, 0x0, 0x0, 0x0)
        /tmp/tmp.W3SuAPdv1O/src/github.com/influxdata/influxdb/tsdb/query_executor.go:276 +0x719
github.com/influxdata/influxdb/tsdb.(*QueryExecutor).planStatement(0xc208030d70, 0x7f39b4208a38, 0xc2181b2dd0, 0xc20b7e0a5e, 0x4, 0x2710, 0x0, 0x0, 0x0, 0x0)
        /tmp/tmp.W3SuAPdv1O/src/github.com/influxdata/influxdb/tsdb/query_executor.go:525 +0x1a4
github.com/influxdata/influxdb/tsdb.(*QueryExecutor).executeStatement(0xc208030d70, 0x0, 0x7f39b4208a38, 0xc2181b2dd0, 0xc20b7e0a5e, 0x4, 0xc20ede13e0, 0x2710, 0xc20ede1380, 0x0, ...)
        /tmp/tmp.W3SuAPdv1O/src/github.com/influxdata/influxdb/tsdb/query_executor.go:649 +0x8f
github.com/influxdata/influxdb/tsdb.func·004()
        /tmp/tmp.W3SuAPdv1O/src/github.com/influxdata/influxdb/tsdb/query_executor.go:172 +0xa1d
created by github.com/influxdata/influxdb/tsdb.(*QueryExecutor).ExecuteQuery
        /tmp/tmp.W3SuAPdv1O/src/github.com/influxdata/influxdb/tsdb/query_executor.go:227 +0x242

```

**Comment by beckettsean** [comment]
@benbjohnson bug in the query engine?

**Comment by benbjohnson** [comment]
@beckettsean Yes, it looks like it's missing `boolean` support too.

**Comment by beckettsean** [comment]
Definite regression from 0.10:

```
InfluxDB shell 0.10.1
> use mydb
Using database mydb
> insert strings value="foo"
> insert strings value="faa"
> select count(value) from strings
name: strings
-------------
time	count
0	2

```

**Comment by jsternberg** [comment]
This was fixed in #5895.
